src/pool/mod.rs

```diff
@@ -16,7 +16,7 @@ pub mod uniswap_v3;
 pub use uniswap_v2::UniswapV2Pool;
 pub use uniswap_v3::UniswapV3Pool;
 
-#[derive(Clone, Copy, Debug)]
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
```

Comment:
Can you remove `PartialEq` & `Eq` traits. Imo these wouldn't have much utility in this context?